dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/condition/ConditionRouter.java

```diff
@@ -262,6 +275,54 @@ private boolean matchCondition(Map<String, MatchPair> condition, URL url, URL pa
         return result;
     }
 
+    /**
+     * analysis the arguments in the rule.
+     * Examples would be like this:
+     * "arguments[0]=1", whenCondition is that the first argument is equal to '1'.
+     * "arguments[1].param=a", whenCondition is that the 'param' attribute in the second argument is equal to 'a'.
+     * "arguments[2].inner.param=b", whenCondition is that the 'param' attribute in the 'inner' attribute of the third argument is equal to 'b'.
+     * @param matchPair
+     * @param invocation
+     * @return
+     */
+    public boolean matchArguments(Map.Entry<String, MatchPair> matchPair, Invocation invocation) {
+        try {
+            // split the rule
+            String key = matchPair.getKey();
+            String[] expressArray = key.split("\\.");
+            String argumentExpress = expressArray[0];
+            final Matcher matcher = ARGUMENTS_PATTERN.matcher(argumentExpress);
+            if (!matcher.find()) {
+                return false;
+            }
+
+            //extract the argument index
+            int index = Integer.parseInt(matcher.group(1));
+            if (index < 0 || index > invocation.getArguments().length) {
+                return false;
+            }
+
+            //extract the argument value
+            Object object = invocation.getArguments()[index];
+            for (int i = 1; i < expressArray.length; i++) {
+                if (object == null) {
+                    return false;
+                }
+                Field field = object.getClass().getDeclaredField(expressArray[i]);
+                field.setAccessible(true);
+                object = field.get(object);
+            }
+
+            if (matchPair.getValue().isMatch((String)object, null)) {

```

Comment:
Shall we adapt those types not match string? Or using toString?

Reply:
Good advice.